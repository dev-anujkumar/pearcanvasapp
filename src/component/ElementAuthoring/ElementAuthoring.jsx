import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {TinyMceEditor} from "../tinyMceEditor"
//import '../../styles/book.scss'
export class ElementAuthoring extends Component {
    constructor(props) {
        super(props);
        this.state = {};
        this.onClick = this.onClick.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onKeyup = this.onKeyup.bind(this);
        this.onFocus = this.onFocus.bind(this);
      }
     
    render() {
        const { type} = this.props
        return(
               <TinyMceEditor type = {type} onFocus = {this.onFocus} onKeyup = {this.onKeyup} onBlur = {this.onBlur} onClick = {this.onClick} />
        )
    }
    onClick() {

    }
    onBlur() {

    }
    onKeyup(){

    }
    onFocus(){

    }
}
ElementAuthoring.defaultProps = {
    type: "element-authoredtext"
}

ElementAuthoring.propTypes = {
      /** Type of element to be rendered */
    type : PropTypes.string.isRequired,
     /** Handler to attach on element click */
    onClick : PropTypes.func,
     /** Handler to attach on element blur */
    onBlur: PropTypes.func,
      /** Handler to attach on element keyup */
    onKeyup:PropTypes.func,
      /** Handler to attach on element focus */
    onFocus:PropTypes.func

}

export default ElementAuthoring