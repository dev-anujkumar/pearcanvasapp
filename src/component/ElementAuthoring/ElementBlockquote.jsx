import React, { Component } from 'react'
import TinyMceEditor from "./../tinyMceEditor"
import './../../styles/ElementAuthoring/ElementAuthoring.css';
import KeyboardWrapper from '../Keyboard/KeyboardWrapper.jsx';
export class ElementBlockquote extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { element, className, model, openGlossaryFootnotePopUp, slateLockInfo, openAssetPopoverPopUp, glossaryFootnoteValue, openMarkedIndexPopUp, markedIndexValue } = this.props
        let blockquoteClass = "blockquoteMarginalia"
        let firstClass = "paragraphNummerEins"
        let secondClass = "blockquoteTextCredit"
        let tempDiv = document.createElement('div');
        tempDiv.innerHTML = model.text;
        let xyz = tempDiv.children[0]?.children[0].outerHTML
        let abc =  tempDiv.children[0]?.children[1].outerHTML
        // let xyz = '<p class="paragraphNummerEins" contenteditable="true"></p>'
        // let abc = '<p class=\"blockquoteTextCredit\" contenteditable=\"true\"></p>'
        let blockquoteElement = 
      <div>
         <blockquote className={blockquoteClass}>
         <KeyboardWrapper enable index={`${this.props.index}-0`}>
                <TinyMceEditor
                    isBlockList={this.props.isBlockList}
                    openAssetPopoverPopUp={openAssetPopoverPopUp}
                    openGlossaryFootnotePopUp={openGlossaryFootnotePopUp}
                    index={`${this.props.index}-0`}
                    elementId={this.props.elementId}
                    element={this.props.element}
                    placeholder={"Type Something..."}
                    className={firstClass}
                    model={xyz}
                    tagName="p"
                    handleEditorFocus={this.props.handleFocus}
                    handleBlur={this.props.handleBlur}
                    slateLockInfo={slateLockInfo}
                    onListSelect={this.props.onListSelect}
                    permissions={this.props.permissions}
                    glossaryFootnoteValue={glossaryFootnoteValue}
                    glossaaryFootnotePopup={this.props.glossaaryFootnotePopup}
                    handleAudioPopupLocation={this.props.handleAudioPopupLocation}
                    handleAssetsPopupLocation={this.props.handleAssetsPopupLocation}
                    showHideType={this.props?.showHideType}
                    parentElement={this.props?.parentElement}
                    openMarkedIndexPopUp={openMarkedIndexPopUp}
                    markedIndexValue={markedIndexValue}
                    parentManifestListItem={this?.props?.parentManifestListItem}
                />
            </KeyboardWrapper>
            <KeyboardWrapper enable index={`${this.props.index}-1`}>
                <TinyMceEditor
                    isBlockList={this.props.isBlockList}
                    openAssetPopoverPopUp={openAssetPopoverPopUp}
                    openGlossaryFootnotePopUp={openGlossaryFootnotePopUp}
                    index={`${this.props.index}-1`}
                    elementId={this.props.elementId}
                    element={this.props.element}
                    placeholder={"Attribution Text"}
                    className={secondClass}
                    model={abc}
                    tagName="p"
                    handleEditorFocus={this.props.handleFocus}
                    handleBlur={this.props.handleBlur}
                    slateLockInfo={slateLockInfo}
                    onListSelect={this.props.onListSelect}
                    permissions={this.props.permissions}
                    glossaryFootnoteValue={glossaryFootnoteValue}
                    glossaaryFootnotePopup={this.props.glossaaryFootnotePopup}
                    handleAudioPopupLocation={this.props.handleAudioPopupLocation}
                    handleAssetsPopupLocation={this.props.handleAssetsPopupLocation}
                    showHideType={this.props?.showHideType}
                    parentElement={this.props?.parentElement}
                    openMarkedIndexPopUp={openMarkedIndexPopUp}
                    markedIndexValue={markedIndexValue}
                    parentManifestListItem={this?.props?.parentManifestListItem}
                />
            </KeyboardWrapper>
            </blockquote>
    </div>
        return blockquoteElement
    }
}


export default ElementBlockquote