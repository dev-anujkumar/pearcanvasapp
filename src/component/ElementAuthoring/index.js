import ElementAuthoring from './ElementAuthoring.jsx';
// import ElementAuthoring from './ElementAuthoring.jsx';

export default ElementAuthoring;
