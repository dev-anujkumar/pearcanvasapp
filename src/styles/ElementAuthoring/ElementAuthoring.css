.Editor {
    padding: 15px !important;
    min-height: 100px;
    flex-grow: 1;
    border: 2px solid #ddd;
    word-break: break-word;
}

.Editor:focus {
    border: 2px solid #149632;
}

.Editor p{
    margin: 0 !important;
}
.element-list .mce-visual-caret{
    display:none;
    cursor:default;
    outline:none;
    border:none;
}