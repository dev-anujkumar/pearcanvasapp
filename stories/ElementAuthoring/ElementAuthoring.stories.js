import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';
import { ElementAuthoring } from '../../src/component/ElementAuthoring/ElementAuthoring';


storiesOf('ElementAuthoring', module)
  .addDecorator(withInfo)
  .add('Paragraph', () => <ElementAuthoring type="element-authoredtext" onClick={action('Add Paragraph')} />, { notes: "Paragraph Element" })
  