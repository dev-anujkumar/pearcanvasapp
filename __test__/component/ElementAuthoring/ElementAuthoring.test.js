import React from 'react';
import { mount } from 'enzyme';
import ReactDOM from 'react-dom';
import ElementAuthoring from '../../../src/component/ElementAuthoring/ElementAuthoring'

describe('Testing Element  component with props', () => {
    const div = document.createElement('div');
    let type = "element-authoredtext";
    let props={
        slateLockInfo:{
            isLocked:false
        }
    }
    const elementAuthoring = mount(<ElementAuthoring {...props} type={type} />);
    let elementAuthoringInstance = elementAuthoring.find('ElementAuthoring').instance();
    it('render Element component ', () => {  
        ReactDOM.render(<ElementAuthoring {...props}/>, div);
    })

})