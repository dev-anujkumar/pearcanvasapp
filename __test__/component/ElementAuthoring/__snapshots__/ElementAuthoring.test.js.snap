// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Element  component with props render Element component  1`] = `
<ElementAuthoring
  type="element-authoredtext"
>
  <TinyMceEditor
    error={null}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyup={[Function]}
    type="element-authoredtext"
  >
    <div
      className="Editor"
    >
      <Editor
        cloudChannel="5"
        init={
          Object {
            "content_style": "@import url(https://fonts.googleapis.com/css?family=Lato);*{border:0;padding:0;margin:0;background:0 0;color:inherit;box-sizing:border-box;background-repeat:no-repeat;background-position:50% 50%;font-weight:400}:focus{outline:0}a{text-decoration:none}b,b *,strong,strong *{font-weight:700}ol,ul{list-style:none}pre{font:inherit}button,input,input:not([type]),input[type=button],input[type=color],input[type=date],input[type=datetime-local],input[type=datetime],input[type=email],input[type=month],input[type=number],input[type=password],input[type=reset],input[type=search],input[type=submit],input[type=tel],input[type=text],input[type=time],input[type=url],input[type=week],select,textarea{font:inherit}@font-face{font-family:aileron;font-style:normal;font-weight:200;src:url(../fonts/aileron-ultra-light.eot?#iefix) format(\\"embedded-opentype\\"),url(../fonts/aileron-ultra-light.woff) format(\\"woff\\"),url(../fonts/aileron-ultra-light.ttf) format(\\"truetype\\")}body{padding:1em}.codepen body{margin:10px 0 0}.codepen body textarea{display:none}.mce-content-body .Wirisformula{margin-top:-5px;}.mce-container textarea{display:inline-block!important}.mce-content-body{font-family:Lato!important;font-size:14px;color:#626262;padding:0 25px 25px}.mce-content-body *{background-position:initial}.mce-content-body h1{font-size:34px;font-family:aileron;font-weight:200;line-height:1.4em;margin:25px 0 15px}.mce-content-body h2{font-size:30px;font-family:aileron;font-weight:200;line-height:1.4em;margin:25px 0 15px}.mce-content-body h3{font-size:26px;font-family:aileron;font-weight:200;line-height:1.4em;margin:25px 0 15px}.mce-content-body h4{font-size:22px;font-family:aileron;font-weight:200;line-height:1.4em;margin:25px 0 15px}.mce-content-body h5{font-size:18px;font-family:aileron;font-weight:200;line-height:1.4em;margin:25px 0 15px}.mce-content-body h6{font-size:14px;font-family:aileron;font-weight:200;line-height:1.4em;margin:25px 0 15px}.mce-content-body p{margin:25px 0}.mce-content-body pre{font-family:monospace}.mce-content-body ol,.mce-content-body ul{margin-left:15px;list-style-position:outside;margin-bottom:20px}.mce-content-body ol li,.mce-content-body ul li{margin-left:10px;margin-bottom:10px;color:#626262}.mce-content-body ul{list-style-type:disc}.mce-content-body ol{list-style-type:decimal}.mce-content-body a[href]{text-decoration:underline}.mce-content-body table{width:100%;border-spacing:0;border-collapse:separate !important;border:1px solid #aaa}.mce-content-body table caption,.mce-content-body table td,.mce-content-body table th{padding:15px 7px;font:inherit}.mce-content-body table td,.mce-content-body table th{border:1px solid #aaa;border-collapse:separate}.mce-content-body table th{font-weight:400;color:#6e6e6e;background-position:100% 100%;background-size:2px 10px;background-repeat:no-repeat}.mce-content-body table th:last-child{background:0 0}.mce-content-body hr{border-top:2px solid #bbb}",
            "fixed_toolbar_container": "#tinymceToolbar",
            "formats": Object {
              "element-authoredtext": Object {
                "block": "p",
                "classes": "paragraphNumeroUno",
              },
              "h1": Object {
                "block": "h1",
                "classes": "heading1NummerEins",
              },
              "h2": Object {
                "block": "h2",
                "classes": "heading2NummerEins",
              },
              "h3": Object {
                "block": "h3",
                "classes": "heading3NummerEins",
              },
              "h4": Object {
                "block": "h4",
                "classes": "heading4NummerEins",
              },
              "h5": Object {
                "block": "h5",
                "classes": "heading5NummerEins",
              },
              "h6": Object {
                "block": "h6",
                "classes": "heading6NummerEins",
              },
            },
            "init_instance_callback": [Function],
            "inline": true,
            "menubar": false,
            "plugins": "lists advlist",
            "selector": ".Editor",
            "setup": [Function],
            "statusbar": false,
            "toolbar": "undo redo| bold italic underline strikethrough removeformat| alignleft aligncenter alignright alignjustify outdent indent numlist bullist | superscript subscript jsplus_special_symbols ",
          }
        }
        onChange={[Function]}
      >
        <div
          id="tiny-react_61769007321566910264173"
        />
      </Editor>
    </div>
  </TinyMceEditor>
</ElementAuthoring>
`;
